src/v/storage/segment.cc

```diff
@@ -96,6 +96,8 @@ ss::future<> segment::close() {
     co_await do_flush();
     co_await do_close();
 
+    vassert(_inflight.empty(), "Closing segment with inflight writes.");
```

Comment:
Is this a problem if the appender continues to live on and complete the inflight writes? Here inflight seems to only be tracking the inflight writes, not owning them.

Reply:
We did acquire the write lock above, we did flush too. Why would we expect outstanding inflight writes?

Reply:
Ahh I didn't notice that you had the fix below.